### PR TITLE
BAEL-6854

### PR DESCRIPTION
**Pull request by cmuramoto**
(feat): Get classpath from ClassLoader

### CODE REVIEW

File: core-java-modules/core-java-classloader/src/test/java/com/baeldung/classloader/GetURLsFromClassloaderTest.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+import com.baeldung.classloader.internal.BasicClasspathResolver;
+import com.baeldung.classloader.internal.InternalClasspathResolver;
+import com.baeldung.classloader.internal.InternalJdkSupport;
+
+class GetURLsFromClassloaderTest {
+
+    final Logger log = LoggerFactory.getLogger(getClass());
+
+    @Test
+    void givenBasicResolver_willNotBeAbleToDetermineURLsFromApplicationClassLoader() {
```
Comment (vinipx) [inline core-java-modules/core-java-classloader/src/test/java/com/baeldung/classloader/GetURLsFromClassloaderTest.java:28]: Make sure we are following the given_when_then test naming - pls, apply for all

**Comment by vinipx** [comment]
@cmuramoto there are still some comments that were not addressed. Could you pls check?

**Comment by cmuramoto** [comment]
> @cmuramoto there are still some comments that were not addressed. Could you pls check?

@vinipx I double-checked the 7 unit tests on the [latest commit](https://github.com/eugenp/tutorials/pull/17506/commits/c17fd7fb86856fe83d16d3c2acbd53324220fe29) and it seems that each one follows the given_when_then convention. Could you kindly detail what else needs to be addressed?

**Comment by vinipx** [comment]
> > @cmuramoto there are still some comments that were not addressed. Could you pls check?
> 
> @vinipx I double-checked the 7 unit tests on the [latest commit](https://github.com/eugenp/tutorials/pull/17506/commits/c17fd7fb86856fe83d16d3c2acbd53324220fe29) and it seems that each one follows the given_when_then convention. Could you kindly detail what else needs to be addressed?

I'm sorry, some of the comments were pending :( my bad
now you can check

File: core-java-modules/core-java-classloader/src/main/java/com/baeldung/classloader/internal/InternalJdkSupport.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+            var bean = ManagementFactory.getRuntimeMXBean();
+            name = bean.getVmName();
+            vendor = bean.getVmVendor();
+            version = bean.getVmVersion();
+        } catch (Exception e) {
+            name = System.getProperty("java.vm.name");
+            vendor = System.getProperty("java.vendor");
+            version = System.getProperty("java.vendor");
+        }
+
```
[the remark this answers is not in the document]
Comment (cmuramoto) [reply core-java-modules/core-java-classloader/src/main/java/com/baeldung/classloader/internal/InternalJdkSupport.java]: done

File: core-java-modules/core-java-classloader/src/main/java/com/baeldung/classloader/internal/InternalJdkSupport.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        if (UCP == null) {
+            throw new UnsupportedOperationException("Program must be initialized with --add-opens java.base/jdk.internal.loader=baeldung.classloader");
+        }
+
+        try {
+            return UCP.get(loader);
+        } catch (Exception e) {
+            throw new InternalError(e);
+        }
+
```
[the remark this answers is not in the document]
Comment (cmuramoto) [reply core-java-modules/core-java-classloader/src/main/java/com/baeldung/classloader/internal/InternalJdkSupport.java]: done

File: core-java-modules/core-java-classloader/src/main/java/com/baeldung/classloader/internal/BasicClasspathResolver.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+    final Logger log = LoggerFactory.getLogger(getClass());
+
+    @Override
+    public void collectClasspath(ClassLoader loader, Set<URL> result) {
+        log.info("Collecting URLs from {}", loader);
+
+        if(loader instanceof URLClassLoader ucl) {
+            var urls = Arrays.asList(ucl.getURLs());
+
```
[the remark this answers is not in the document]
Comment (cmuramoto) [reply core-java-modules/core-java-classloader/src/main/java/com/baeldung/classloader/internal/BasicClasspathResolver.java:23]: done

**Pull request merged**